yapio.c

```diff
@@ -1251,9 +1268,36 @@ yapio_alloc_buffers(const yapio_test_group_t *ytg)
         yapio_destroy_buffers_and_abort();
 }
 
+/* read input md file to initialize yapioMyRank's md array */
+static void
+yapio_initialize_source_md_buffer_from_file(const yapio_test_group_t *ytg)
+{
+    /* each rank reads its own md file '{filename}-md-{rank}' */
+    char md_file[PATH_MAX + 10];
+    snprintf(md_file, PATH_MAX, "%s-md-%d", yapioTestFileName, yapioMyRank);
+
+    int nblks_per_rank = ytg->ytg_num_blks_per_rank;
+
+    FILE *file = fopen(md_file, "r");
+
+    if (file)
+    {
+        fread(yapioSourceBlkMd, sizeof(yapio_blk_md_t), nblks_per_rank, file);
+        fclose(file);
+    }
+    else
+        printf("Unable to read initial metadata state\n");
+}
+
 static void
 yapio_initialize_source_md_buffer(const yapio_test_group_t *ytg)
 {
+    if (yapioInitFromMdFile)
+    {
+        yapio_initialize_source_md_buffer_from_file(ytg);
+        return;
```

Comment:
you should not be able to continue if you cannot init the metadata even more if you start with the read operation

Reply:
the program would now crash in yapio_initialize_source_md_buffer_from_file() in this case